lib/App/Yath/Plugin.pm

```diff
@@ -23,3 +23,203 @@ sub claim_file {}
 sub inject_run_data {}
 
 1;
+
+__END__
+
+=pod
+
+=encoding UTF-8
+
+=head1 NAME
+
+App::Yath::Plugin - Parent of App::Yath plugins
+
+=head1 SYNOPSIS
+
+  package App::Yath::Plugin::SOME_PLUGIN;
+  use strict;
+  use warnings;
+
+  use parent 'App::Yath::Plugin';
+
+  sub find_files {
+    my @files;
+    # ...
+
+    return \@files;
+  }
+
+  1;
+
+  # which can then be used when invoking `yath`:
+
+  $ yath -pSOME_PLUGIN
+
+
+=head1 DESCRIPTION
+
+L<App::Yath>'s processing can be hooked into at specific points
+by plugins. These plugins can extend or change its default processing.
+
+This module provides an empty skeleton plugin. It's meant to be used
+as a parent class for any custom plugins.
+
+=head2 Instantiation
+
+In case a plugin has a C<new> function, it is assumed to be a constructor
+and an instance will be created using any parameters provided on the command
+line:
+
+  $ yath -pSOME_PLUGIN=abc,1,dev,delta
+
+  # causes the 'new' function to be invoked as:
+  #  App::Yath::Plugin::SOME_PLUGIN->new( abc => 1, dev => 'delta' );
```

Comment:
You might make it clear that the arguments are passed as a list, so it is ok for a plugins new() method to take a list of args instead of named/hash-style args.